test/test_utils.go

```diff
@@ -107,3 +114,70 @@ func newRPCClient(
 	client := workflowserviceclient.New(dispatcher.ClientConfig(serviceName))
 	return &rpcClient{Interface: client, dispatcher: dispatcher}, nil
 }
+
+// stringMapPropagator propagates the list of keys across a workflow,
+// interpreting the payloads as strings.
+// BORROWED FROM 'internal' PACKAGE TESTS.
```

Comment:
Help remove the class from internal package and change all the reference to this one?

Reply:
sure

Reply:
thinking about it some more....is it a good approach to have code in an implementation package (internal) reference and rely on a struct defined in the test package?

also, I'm thinking this can introduce an import cycle.
test  > workflow > internal > test. How about making stringMapPropagator public? or having a "common" package used by both the internal and test packages?

Reply:
My understanding is that the reference in internal package is also for testing purpose, right?
It does not seem that there's a common package outside of internal. So I am ok leaving this as is for now.

Reply:
Ok, so I've fixed your other 2 comments (newline and error message).
Merge at will :)

Thanks, very happy to contribute!